packages/lodestar/src/eth1/impl/ethers.ts

```diff
@@ -101,7 +101,8 @@ export class EthersEth1Notifier implements IEth1Notifier {
     }
     this.eth1Source = pushable<Eth1EventsBlock>();
     // no need await
-    this.startProcessEth1Blocks();
+    // ### Todo: Unknown promise duration
+    await this.startProcessEth1Blocks();
```

Comment:
What is the reason of the `no need await` comment? What happens if this is awaited?

Reply:
again endless loop, it would never resolved so this method wouldn't return eth1 block generator

Reply:
@wemeetagain But this approach will create the same situation of unhandled rejected promises if `longRunningThing` throws inside of a setTimeout

Reply:
I guess longRunningThing should explicitly have a try/catch.

I'm just thinking of our needs this way:
- coherent/standard strategy for handling daemon-like subprocesses across modules
- standard behavior for start/stop across modules

Reply:
If this function never resolves (or takes a long time) it means that the `start` method never resolves either?

https://github.com/ChainSafe/lodestar/blob/bd8adf267acebd0a93efa693be8fcec538384431/packages/lodestar/src/eth1/impl/ethers.ts#L80-L85

 If yes, then `Chain.start()` will never resolve either?
https://github.com/ChainSafe/lodestar/blob/bd8adf267acebd0a93efa693be8fcec538384431/packages/lodestar/src/chain/chain.ts#L137
https://github.com/ChainSafe/lodestar/blob/bd8adf267acebd0a93efa693be8fcec538384431/packages/lodestar/src/chain/chain.ts#L152-L153

Reply:
right, this specific case might be resolved with #1103

Reply:
Got it. But for now, how doesn't this stall the node's start process?

Reply:
It does stall the nodes start process. IMO the takeaway of this PR should be:
1. Determine root causes of problems here (eg: needing an error-handling strategy for long-running processes)
2. Document problem areas in the code (there are a handful here)
3. Eslint rule in place to prevent further unhandled rejections

In the problem cases, just add an eslint-disable line for now. Don't try to add hacks that just mask the problem, because a real solution is likely bigger than can be solved here.

Reply:
Opened followup issue https://github.com/ChainSafe/lodestar/issues/1196